src/Application.php

```diff
@@ -320,8 +320,12 @@ public function configureAndRegisterCommands(InputInterface $input, OutputInterf
         $discovery = $this->commandDiscovery();
         $commandClasses = $discovery->discover($commandfileSearchpath, '\Drush');
         $commandClasses[] = \Consolidation\Filter\Hooks\FilterHooks::class;
+        $commandsFromConfig = $this->commandsFromConfiguration();
+        if ($commandsFromConfig) {
+            @trigger_error("Discovering global commands by specifying them in 'drush.commands' config is deprecated and will be removed in Drush 11.0.0. See docs/commands.md on how to create auto-discoverable global commands.", E_USER_DEPRECATED);
```

Comment:
I am wondering if `trigger_error` is too noisy. Maybe we should deprecate with `trigger_error` in Drush 11, and remove in Drush 12? I know that's a long long ways off, but folks are very slow at upgrading Drush commands in modules.

Reply:
>  I know that's a long long ways off, but folks are very slow at upgrading Drush commands in modules.

This is something that leads to exactly the opposite conclusion: If they're slow to adopt changes, then we should deprecate as soon as possible, giving them enough time to take action. We can change the removal version from 11.0.0 to 12.0.0 but the deprecation should occur now. Deprecating doesn't mean the command doesn't work. Also note that this is a muted deprecation ` @trigger_error(...)`, so there should be no noise in console. Normally code deprecation tests should catch this error but that's fine.

Reply:
Anyway, deprecation is postponed.